sandy/fy.py

```diff
@@ -350,13 +352,151 @@ def apply_qmatrix(self, zam, energy, decay_data):
+        """
+        “a priori” calculated value for the vector of GLS.
+
+        Parameters
+        ----------
+        zam : `int`
+            ZAM number of the material to which calculations are to be
+            applied.
+        e : `float`
+            Energy to which calculations are to be applied.
+        decay_data : `DecayData`
+            Container of radioactive nuclide data for several isotopes.
+        kind : `str`
+            Keyword for obtaining sensitivity.
+
+        Raises
+        ------
+        TypeError
+            The kind is not implemented in the method.
+
+        Returns
+        -------
+        `pandas.Series`
+            “a priori” calculated vector.
+
+        """
+        fy_data = self.filter_by('ZAM', zam).filter_by("E", e)
+        if kind == 'cumulative':
+            y = fy_data.apply_qmatrix(zam, e, decay_data).data
+            # Apropiate format
+            y = y[y.MT == 459].set_index('ZAP').FY
+        else:
+            raise TypeError('The kind introduced is not valid')
+        return y  # pd.Series with the vector of the calculus.
+
+    def gls_update(self, zam, e, decay_data, y, Vy, kind='cumulative',
+                   threshold=None):
+        """
+        Update IFY for a given zam, energy, decay_data and new CFY information.
+
+        Parameters
+        ----------
+        zam : `int`
+            ZAM number of the material to which calculations are to be
+            applied.
+        e : `float`
+            Energy to which calculations are to be applied.
+        decay_data : `sandy.DecayData`
+            Radioactive nuclide data for several isotopes.
+        y : 1D iterable
+            New value of the vector.
+        Vy : 2D iterable
+            Extra Covariance matrix (MXM).
+        kind : `str`, optional
+            Keyword for obtaining sensitivity. The
+            default is 'cumulative'.
+        threshold : `int`, optional
+            Optional argument to avoid numerical fluctuations or
+            values so small that they do not have to be taken into
+            account. The default is None.
+        index_columns : `str`, optional
+            Keyword to determine if the sensitivity is the transposed one.
+            The default is None.
+
+        Returns
+        -------
+        `FY`
+            IFY updated with GLS for a given zam, energy, decay_data and
+            new CFY information.
+
+        """
+        data = self
+        # Filter FY data:
+        fy_data = data.filter_by('ZAM', zam)\
+                      .filter_by("E", e).data\
+                      .set_index('ZAP')[['MT', 'FY', 'DFY']]
+        # Divide the data:
+        IFY = fy_data.query('MT==454').FY
+        mask = (data.data.ZAM == zam) & (data.data.MT == 454) & (data.data.E == e)
+        DIFY = fy_data.query('MT==459').DFY
+        # Find the GLS varibles:
+        S = _gls_setup(decay_data, kind)
+        x_calc = data._gls_calc(zam, e, decay_data, kind="cumulative")
```

Comment:
let's use some common nomenclature...and please adjust also `gls.py`.

MODEL
x : model input
y : model output
y=f(x) : model

PRIOR
x_prior : 1D prior data, e.g. IFY
Vx_prior : 2D covariance matrix of x_prior
S : 2D sensitivity of the model y=f(x)
y_calc : 1D calculated output using S.dot(x_prior), e.g. calculated CFY
Vy_calc : 2D calculated output using S.T.dot(Vx_prior.dot(S))

EXTRA
y_extra : 1D extra info on output
Vy_extra : 2D covariance matrix for y_extra


you often use `x` for the model output (like in this case), and then it gets messy